crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -181,16 +207,121 @@ where
     }
 }
 
+trait HasIndexation<'a> {
```

Comment:
I'm confused by this `trait` name. I'd expect a method that returns a `bool` to say if it has indexation. Instead it has 3 methods.

Reply:
Yes, this trait had a different form initially, and that's the leftover. Currently I'm trying to simplify the code around this trait, I'll then make sure it is named properly.

Reply:
I decided to split this trait in two for the sake of separation of concerns. Also, I though about hopefully better name.
https://github.com/FuelLabs/fuel-core/pull/2383/commits/91e8abc6c8cd67a22b0119bf9f54c25fd4e6b168

Reply:
This trait and corresponding boilerplate code has been removed in favor of non-generic functions inside the `indexation.rs` module.